### PR TITLE
Create ShippingCheckoutComponent.php

### PR DESCRIPTION
**Pull request by slievr**
Please advise on how this could be cleaned up

### CODE REVIEW

File: code/components/ShippingCheckoutComponent.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	public function validateData(Order $order, array $data) {
+		$result = new ValidationResult();
+		if(!isset($data['ShippingMethodID'])){
+			$result->error("Shipping method not provided", "ShippingMethod");
+			throw new ValidationException($result);
+		}		
+		
+		if(!ShippingMethod::get()->byID($data['ShippingMethodID'])){
+		 	$result->error("Gateway not supported", "ShippingMethod");
```
Comment (markguinn) [inline code/components/ShippingCheckoutComponent.php]: Is this the correct error message?

Comment (slievr) [inline code/components/ShippingCheckoutComponent.php]: Ah, no it should be 'Shipping Method does not exist' instead of 'gateway not supported'

Comment (markguinn) [inline code/components/ShippingCheckoutComponent.php]: This all looks good. If you could fix that message and squash the two commits I'm happy to merge this. Thanks for your work!

**Comment by slievr** [comment]
Hello, I've made that change. Sorry for the lateness I was out of action for a week.

**Comment by markguinn** [comment]
Thanks, @slievr. Would you mind squashing those two commits together for cleanliness?

**Comment by markguinn** [comment]
@slievr would you like me to go ahead and squash the two commits for you so we can get this merged?

**Comment by slievr** [comment]
@markguinn Please, and thanks.

**Comment by wilr** [comment]
Fixed in 6fac2617dc4e955f09b08190903097122bd09dc5

**Pull request closed without merging**